src/dotnet/HQ.Server/Services/HolidayServiceV1.cs

```diff
@@ -205,5 +212,62 @@ public async Task<Result<ImportHolidayV1.Response>> ImportHolidayV1(ImportHolida
             Updated = updated
         };
     }
+    public async Task BackgroundAutoGenerateHolidayTimeEntryV1(CancellationToken ct)
+    {
+        var today = DateOnly.FromDateTime(DateTime.UtcNow);
+        var holidays = _context.Holidays
+        .AsNoTracking()
+        .AsQueryable();
+        var jurisdicitons = Enum.GetValues(typeof(Jurisdiciton));
+        foreach (Jurisdiciton jurisdiciton in jurisdicitons)
+        {
+            // retrieve upcoming holiday based on today's date
+            var upcomingHoliday = await holidays.Where(t => t.Date >= today && t.Jurisdiciton == jurisdiciton).FirstOrDefaultAsync(ct);
+            if (upcomingHoliday == null)
```

Comment:
We need to loop through all holidays in the current booking period week, Christmas week for example we have the whole week off so we would have 5 holiday entries for Christmas.